apps/schedules/lib/hours_of_operation.ex

```diff
@@ -146,6 +199,52 @@ defmodule Schedules.HoursOfOperation do
     {:error, :timeout}
   end
 
+  defp special_service_departures({:error, _} = error, _, _) do
+    error
+  end
+
+  defp special_service_departures(special_service_responses, headsigns, description) do
+    with {:ok, special_service_depature_maps} <-
+           special_service_departures_parser(special_service_responses, headsigns, description) do
+      {
+        :ok,
+        special_service_depature_maps
+      }
+    end
+  end
+
+  # These should come in chunks of 2, anything else and its bad data
+  defp special_service_departures_parser(
+         [{:ok, %JsonApi{data: data_0}}, {:ok, %JsonApi{data: data_1}} | tail],
+         headsigns,
+         description
+       ) do
+    date_key = get_date_string(List.first(data_0))
+
+    direction_tuple =
+      {departure(data_0, headsigns, description), departure(data_1, headsigns, description)}
+
+    with {:ok, special_service_departures_map} <-
+           special_service_departures_parser(tail, headsigns, description) do
+      {:ok, Map.put(special_service_departures_map, date_key, direction_tuple)}
+    end
+  end
+
+  # Reached the end of the array, return the map we build on
+  defp special_service_departures_parser([], _, _) do
+    {:ok, %{}}
+  end
+
+  defp special_service_departures_parser(_, _, _) do
+    {:error, "Unexpected special service hours data"}
+  end
+
+  defp get_date_string(data) do
+    departure_time = Map.get(data.attributes, "departure_time")
+    # Get the date part of the date time string
+    List.first(String.split(departure_time, "T"))
```

Comment:
**suggestion**: Schedules may have a null `departure_time`! Which would theoretically trip this up, but to be honest I haven't run into it while I've been QAing this. But would still be a good idea to account for that here.